marketing/sailthruScriptJob.groovy

```diff
@@ -0,0 +1,128 @@
+Map secretMap = [:]
+out.println('Parsing secret YAML file')
+try {
+    /* Parse k:v pairs from the secret file referenced by secretFileVariable */
+    String contents = new File("${EDX_SAILTHRU_SYNC_CONTENT}").text
+    Yaml yaml = new Yaml()
+    secretMap = yaml.load(contents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    assert jobConfig.containsKey('oauth_host')
+    assert jobConfig.containsKey('oauth_key')
+    assert jobConfig.containsKey('oauth_secret')
+    assert jobConfig.containsKey('sailthru_key')
+    assert jobConfig.containsKey('sailthru_secret')
+    assert jobConfig.containsKey('content_api_url')
+    assert jobConfig.containsKey('lms_url')
+    assert jobConfig.containsKey('report_email')
+    assert jobConfig.containsKey('type')
+    assert jobConfig.containsKey('fixups')
+    assert jobConfig.containsKey('mode')
+    assert jobConfig.containsKey('requirements')
+    assert jobConfig.containsKey('scriptPath')
+    assert jobConfig.containsKey('githubRepoUrlSegment')
+    assert jobConfig.containsKey('jobName')
+    assert jobConfig.containsKey('notificationEmail')
+
+    job(jobConfig['jobName']) {
+
+        requirements = "sudo pip install -r " + jobConfig['requirements']
+        sync_content = "python " + jobConfig['scriptPath'] + " --oauth_host "+ jobConfig['oauth_host'] + 
+                  " --oauth_key " + jobConfig['oauth_key'] + " --oauth_secret " + jobConfig['oauth_secret'] + 
+                  " --sailthru_key " + jobConfig['sailthru_key'] + " --sailthru_secret " + jobConfig['sailthru_secret'] +
+                  " --content_api_url " + jobConfig['content_api_url'] + " --lms_url " + jobConfig['lms_url'] +
+                  " --report_email " + jobConfig['report_email'] + " --fixups=" + jobConfig['fixups'] +
+                  " --type " + jobConfig['type'] + " " + jobConfig['mode']
+
+        properties {
+              githubProjectUrl(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['githubRepoUrlSegment'])
+        }
+
+        logRotator JENKINS_PUBLIC_LOG_ROTATOR() //Discard build after 14 days
+
+        label('master') //restrict to jenkins-worker
+
+        checkoutRetryCount(5)
+
+        scm {
+
+           git { //using git on the branch and url, clone, clean before checkout
+                remote {
+                    url(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['githubRepoUrlSegment'] + '.git')
+                }
+                branch('*/master')
+                browser()
+            }
+        }
+
+        steps {
+            shell( "sudo rm -rf /usr/local/lib/python2.7/dist-packages/requests*")
```

Comment:
@umar-nawaz So what you are saying is that ShiningPanda is still not configured correctly on your local Jenkins instance? I think you should spend the time to look into how ShiningPanda should be configured, rather than having this hack here. I can try to help more tomorrow if you are still stuck.